src/results_cache.py

```diff
@@ -35,6 +35,24 @@ def set_result(self, input_id, predicate, qualifiers, source_input, caller, work
         key = self.get_query_key(input_id, predicate, qualifiers, source_input, caller, workflow)
 
         self.redis.set(key, gzip.compress(json.dumps(final_answer).encode()))
+
+    def get_lookup_query_key(self, workflow, query_graph):
+        keydict = {'workflow': workflow, 'query_graph': query_graph}
+        return json.dumps(keydict, sort_keys=True)
+
+    def get_lookup_result(self, workflow, query_graph):
+        key = self.get_lookup_query_key(workflow, query_graph)
+        result = self.redis.get(key)
```

Comment:
I'm still undecided on this, but would it make sense to use a different redis db for lookup queries? That way they're somewhat separated from the inferred queries.

Reply:
That's something I'd considered, but I wasn't sure how it would affect deployment. I feel like it might make it a bit complicated.

Reply:
It wouldn't change anything. Redis has multiple internal dbs it can use. We'd be using the same redis instance.